arwen/host/execution.go

```diff
@@ -303,13 +299,14 @@ func (host *vmHost) execute(input *vmcommon.ContractCallInput) error {
 		return arwen.ErrFunctionRunError
 	}
 
+	if !result.IsVoid() {
```

Comment:
It might be too restrictive to throw an error if the called function returns a value. Better to simply ignore `result` altogether?

Reply:
I think it is better to signal to the developer that the return value is not allowed than ignoring it and making them wonder why what they thought should work doesn't work.

It should only be an issue for developers going 100% raw - even for C I'm thinking we should come up with some macro magic.